### PR TITLE
Issue 192

### PR DESCRIPTION
**Pull request by Que-tin**

### CODE REVIEW

File: src/props.animations.js [hunk 1]
```diff
@@ -77,20 +77,20 @@ export default {
...
+  "--animation-ping": "ping var(--ping-duration, 1ms) var(--ease-out-3) infinite", 
   "--animation-ping-@": `
 @keyframes ping {
   90%, 100% {
     transform: scale(2);
     opacity: 0;
   }
 }`,
-  "--animation-blink": "blink 1s var(--ease-out-3) infinite", 
+  "--animation-blink": "blink var(--blink-duration, 1ms) var(--ease-out-3) infinite",
```
Comment (argyleink) [inline src/props.animations.js:93]:
```suggestion
  "--animation-blink": "blink var(--blink-duration, 2s) var(--ease-out-3) infinite", 
```

for this one, let's go a bit more custom and double the duration, so it's a non-threatening but still effective blink. i felt stepped animations were much were than just allowing the blink crossfade.

File: src/props.animations.js [hunk 2]
```diff
@@ -100,19 +100,19 @@ export default {
     opacity: .5
   }
 }`,
-  "--animation-float": "float 3s var(--ease-in-out-3) infinite", 
+  "--animation-float": "float var(--float-duration, 1ms) var(--ease-in-out-3) infinite",
```
Comment (argyleink) [inline src/props.animations.js:103]:
```suggestion
  "--animation-float": "float 3s var(--float-ease, steps(2)) infinite", 
```

got a `steps(2)` here, breaking the pattern!

File: src/props.animations.js [hunk 3]
```diff
@@ -100,19 +100,19 @@ export default {
...
   }
 }`,
-  "--animation-float": "float 3s var(--ease-in-out-3) infinite", 
+  "--animation-float": "float var(--float-duration, 1ms) var(--ease-in-out-3) infinite", 
   "--animation-float-@": `
 @keyframes float {
   50% { transform: translateY(-25%) }
 }`,
-  "--animation-bounce": "bounce 2s var(--ease-squish-2) infinite", 
+  "--animation-bounce": "bounce var(--bounce-duration, 1ms) var(--ease-squish-2) infinite",
```
Comment (argyleink) [inline src/props.animations.js:108]:
```suggestion
  "--animation-bounce": "bounce var(--bounce-duration, 4s) var(--bounce-ease, steps(2)) infinite", 
```

the first double

File: src/props.animations.js [hunk 4]
```diff
@@ -100,19 +100,19 @@ export default {
...
-  "--animation-bounce": "bounce 2s var(--ease-squish-2) infinite", 
+  "--animation-bounce": "bounce var(--bounce-duration, 1ms) var(--ease-squish-2) infinite", 
   "--animation-bounce-@": `
 @keyframes bounce {
   25% { transform: translateY(-20%) }
   40% { transform: translateY(-3%) }
   0%, 60%, 100% { transform: translateY(0) }
 }`,
-  "--animation-pulse": "pulse 2s var(--ease-out-3) infinite", 
+  "--animation-pulse": "pulse var(--pulse-duration, 1ms) var(--ease-out-3) infinite",
```
Comment (argyleink) [inline src/props.animations.js:115]:
```suggestion
  "--animation-pulse": "pulse 2s var(--pulse-ease, steps(2)) infinite", 
```

File: src/props.animations.js [hunk 5]
```diff
@@ -68,7 +68,7 @@ export default {
   60% { transform: translateX(-5%) }
   80% { transform: translateX(5%) }
 }`,
-  "--animation-shake-y": "shake-y .75s var(--ease-out-5)", 
+  "--animation-shake-y": "shake-y var(--shake-duration, 1ms) var(--ease-out-5)",
```
Comment (argyleink) [inline src/props.animations.js:71]:
```suggestion
  "--animation-shake-y": "shake-y var(--shake-duration, 3s) var(--shake-easing, steps(2))", 
```

File: src/props.animations.js [hunk 6]
```diff
@@ -9,57 +9,57 @@ export default {
...
   from { transform: translateX(-100%) }
 }`,
-  "--animation-slide-in-left": "slide-in-left .5s var(--ease-3)", 
+  "--animation-slide-in-left": "slide-in-left var(--slide-duration, 1ms) var(--ease-3)", 
   "--animation-slide-in-left-@": `
 @keyframes slide-in-left {
   from { transform: translateX(100%) }
 }`,
-  "--animation-shake-x": "shake-x .75s var(--ease-out-5)", 
+  "--animation-shake-x": "shake-x var(--shake-duration, 1ms) var(--ease-out-5)",
```
Comment (argyleink) [inline src/props.animations.js:62]:
```suggestion
  "--animation-shake-x": "shake-x var(--shake-duration, 3s) var(--shake-easing, steps(2))", 
```

File: src/props.animations.js [hunk 7]
```diff
@@ -9,57 +9,57 @@ export default {
...
   from { transform: translateY(-100%) }
 }`,
-  "--animation-slide-in-right": "slide-in-right .5s var(--ease-3)", 
+  "--animation-slide-in-right": "slide-in-right var(--slide-duration, 1ms) var(--ease-3)", 
   "--animation-slide-in-right-@": `
 @keyframes slide-in-right {
   from { transform: translateX(-100%) }
 }`,
-  "--animation-slide-in-left": "slide-in-left .5s var(--ease-3)", 
+  "--animation-slide-in-left": "slide-in-left var(--slide-duration, 1ms) var(--ease-3)",
```
Comment (argyleink) [inline src/props.animations.js:57]: havent looked into the solution yet, but these `slide-in` animations dont looks right with reduced motion turned on. we can definitely have them behave more like `slide-out` animations, they're nice with reduced motion. can you peek into these ones? 🙂

File: build/props.js [hunk 8]
```diff
@@ -18,7 +18,14 @@ import {toJSON} from './to-json.js'
...
 
-const [,,prefix,useWhere] = process.argv
+let prefix = process.argv[2];
+const useWhere = process.argv[3]
+
+// On windows devices the prefixs is "''" if not set
+if(prefix === "''") {
+  prefix = undefined;
+}
+
```
Comment (argyleink) [inline build/props.js:28]: don't need this as part of this PR sorry!

**Review by argyleink — CHANGES_REQUESTED** [review]
well this turned out to be a bit more than anticipated! cool news is I think this is really going to up the bar for animation libraries and reduced motion, thanks so much for kicking this off.

there's comments all over the place, mostly me realizing stuff and offering some code suggestions. 11 comments sounds like a lot lol, but i think it'll go quick! 🙏🏻 

here's all of them turned on, aka showing them in their worst light but also an overview

https://user-images.githubusercontent.com/1134620/155270135-b290a6da-4564-432e-a188-33d900c419ec.mp4

File: src/props.animations.js (hunk 7 above)
Comment (Que-tin) [reply src/props.animations.js:57]: Just checked. I agree with you they don't look right, but they are indeed right.

As `slide in`'s `from` state is not on the same spot as the element itself it looks kind of broken. It does basically the same as the `slide-out`'s just the other way around. As the play JavaScript removes the animation only after a while the ` slide-out`'s stays where it is before jumping back, the ` slide-in` does the same by staying on the spot where it started. We don't see the flickering from the `1ms` on the slide-out as it stays where it is upon starting the animation.

The flickering would be removed when setting the animation timing for reduced motion to `0ms`. I wrote a longer explanaition / question regarding this topic inside of discord.

**Comment by Que-tin** [comment]
Done. Somone needs to check the reduced motion timings. Especially shake & bounce are behaving a bit weird.

**Comment by argyleink** [comment]
did some polish and fixed an animation, made a PR to your branch! https://github.com/Que-tin/open-props/pull/1

**Comment by Que-tin** [comment]
@argyleink Checked and tested everything, looks good to me 👍🏻😊

Your PR also got merged.

**Comment by argyleink** [comment]
preview URL https://op-reduced-animations.netlify.app/#animations

**Comment by argyleink** [comment]
review #1:

> I agree but I can only speak for myself: most of the time I'm okay with small animations (like color fade), it's mostly paralax and things moving around too much that trigger me ^^
And spinning hahaha spinning is NOOOO good for my brain 😄
haa the spining even with motion reduced is a big NOPE hahaa
The step by step animations feel strange without transition, to be honnest. It's a little bit hard to give you feedback since it also depends on how those are used in a page
The bounce is also a big nope at the moment even in the reduce motion, but the pulse is okay. So I think it's the size of the change and the speed as well that impacts
but again, it might depend on the level of sensitivity to motion sickness

**Comment by argyleink** [comment]
i still think about these efforts and where they should go 🙂

**Comment by Que-tin** [comment]
Closed due to the progress pn #341

**Pull request closed without merging**